MimeUtility.cs

```diff
@@ -6,41 +6,31 @@
 namespace MimeMapping
 {
     /// <summary>
-    /// MIME type utility to lookup by file extension
+    ///     MIME type utility to lookup by file extension
     /// </summary>
     public static class MimeUtility
     {
         /// <summary>
-        /// The "octet-stream" subtype is used to indicate that a body contains arbitrary binary data.
-        /// See <a href="https://www.iana.org/assignments/media-types/application/octet-stream">application/octet-stream</a>
+        ///     The "octet-stream" subtype is used to indicate that a body contains arbitrary binary data.
+        ///     See <a href="https://www.iana.org/assignments/media-types/application/octet-stream">application/octet-stream</a>
         /// </summary>
         public const string UnknownMimeType = "application/octet-stream";
 
-        static Lazy<ReadOnlyDictionary<string, string>> _lazyDict = new Lazy<ReadOnlyDictionary<string, string>>(
-            () => new ReadOnlyDictionary<string, string>(KnownMimeTypes.ALL_EXTS.Value.ToDictionary(e => e, e => KnownMimeTypes.LookupType(e)))
-        );
+        private static readonly Lazy<ReadOnlyDictionary<string, string>> LazyDict = new Lazy<ReadOnlyDictionary<string, string>>(() => new ReadOnlyDictionary<string, string>(KnownMimeTypes.ALL_EXTS.Value.ToDictionary(e => e, KnownMimeTypes.LookupType)));
 
         /// <summary>
-        /// Dictionary of all available types (lazy loaded on first call)
+        ///     Dictionary of all available types (lazy loaded on first call)
         /// </summary>
-        public static ReadOnlyDictionary<string, string> TypeMap => _lazyDict.Value;
+        public static ReadOnlyDictionary<string, string> TypeMap => LazyDict.Value;
 
         /// <param name="file">The file extensions (ex: "zip"), the file name, or file path</param>
         /// <returns>The mime type string, returns "application/octet-stream" if no known type was found</returns>
         public static string GetMimeMapping(string file)
         {
-            string fileExtension;
-            if (file.Contains("."))
-            {
-                fileExtension = Path.GetExtension(file).Substring(1);
-            }
-            else
-            {
-                fileExtension = file;
-            }
-            string mimeType = KnownMimeTypes.LookupType(fileExtension.ToLowerInvariant());
+            if (string.IsNullOrEmpty(file)) return UnknownMimeType;
```

Comment:
Considering typical use cases, I think empty strings can be expected, and returning `UnknownMimeType` immediately makes sense. But for null I think we should be throwing an argument null exception.

Reply:
I was thinking that as well. I originally put:
```
if (string.IsNullOrEmpty(file)) throw new ArgumentNullException(nameof(file));
```
That seems reasonable? 

Thanks for putting together this package. I was previously using the AspnetCore StaticFiles package just to use `FileExtensionContentTypeProvider` to do the same thing your dictionary provided.

Reply:
That null check and exception look good. 

I imagine many of the file inputs come from arbitrary web clients, and I think developers would be more surprised by an empty string throwing and exception rather than returning unknown.